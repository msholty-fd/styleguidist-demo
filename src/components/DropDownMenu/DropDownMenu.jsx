import React from 'react';
import DropDownMenu from 'material-ui/DropDownMenu';

import list from 'styleguide/components/list';
import dropDownStyles from 'styleguide/components/dropDownMenu';

const propTypes = {
  ...DropDownMenu.propTypes,
  children: React.PropTypes.array,
};

const getStyles = props => ({
  list: Object.assign({}, list, props.listStyle),
  root: Object.assign({}, dropDownStyles.root, props.style),
  underline: Object.assign({}, dropDownStyles.underline, props.underlineStyle),
  label: Object.assign({}, dropDownStyles.label, props.labelStyle),
  icon: Object.assign({}, dropDownStyles.icon, props.iconStyle),
});

const SharpspringDropDown = (props) => {
  const styles = getStyles(props);

  return (
    <DropDownMenu
      {...props}
      style={styles.root}
      listStyle={styles.list}
      underlineStyle={styles.underline}
      labelStyle={styles.label}
      iconStyle={styles.icon}
    >
      {props.children}
    </DropDownMenu>
  );
};

SharpspringDropDown.propTypes = propTypes;

export default SharpspringDropDown;
