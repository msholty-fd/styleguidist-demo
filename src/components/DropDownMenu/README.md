Example:
```
const MenuItem = require('material-ui').MenuItem;

<DropDownMenu value={1}>
  <MenuItem value={1} primaryText="Never" />
  <MenuItem value={2} primaryText="Every Night" />
  <MenuItem value={3} primaryText="Weeknights" />
  <MenuItem value={4} primaryText="Weekends" />
  <MenuItem value={5} primaryText="Weekly" />
</DropDownMenu>
```
