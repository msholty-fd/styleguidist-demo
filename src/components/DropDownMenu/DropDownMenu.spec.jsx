import React from 'react'
import renderer from 'react-test-renderer'

import MenuItem from 'material-ui/MenuItem'
import DropDownMenu from './DropDownMenu'
import { describe, it, expect } from 'global'

jest.mock('material-ui/DropDownMenu', () => 'DropDownMenu')
jest.mock('material-ui/MenuItem', () => 'MenuItem')

const component = (
  <DropDownMenu
    value={3}
    style={{
      marginRight: 10
    }}
    maxHeight={300}
  >
    <MenuItem value={1} primaryText='Never' />
    <MenuItem value={2} primaryText='Every Night' />
    <MenuItem value={3} primaryText='Weeknights' />
    <MenuItem value={4} primaryText='Weekends' />
    <MenuItem value={5} primaryText='Weekly' />
  </DropDownMenu>
)

describe('<DropDownMenu />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(component)
    const tree = renderedComponent.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
