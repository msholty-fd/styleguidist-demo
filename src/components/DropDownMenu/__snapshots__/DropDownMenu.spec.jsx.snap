// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropDownMenu /> should match the snapshot 1`] = `
<DropDownMenu
  iconStyle={
    Object {
      "height": 40,
      "padding": "8px 0 0 20px",
      "top": 0,
      "width": 40,
    }
  }
  labelStyle={
    Object {
      "color": "#ffffff",
      "lineHeight": "40px",
      "maxWidth": 150,
      "paddingRight": 40,
    }
  }
  listStyle={
    Object {
      "borderRadius": 5,
      "padding": "10px 0",
    }
  }
  maxHeight={300}
  style={
    Object {
      "background": "#a8b0b0",
      "borderRadius": 3,
      "height": 40,
      "lineHeight": 40,
      "marginRight": 10,
    }
  }
  underlineStyle={
    Object {
      "border": 0,
    }
  }
  value={3}
>
  <MenuItem
    primaryText="Never"
    value={1}
  />
  <MenuItem
    primaryText="Every Night"
    value={2}
  />
  <MenuItem
    primaryText="Weeknights"
    value={3}
  />
  <MenuItem
    primaryText="Weekends"
    value={4}
  />
  <MenuItem
    primaryText="Weekly"
    value={5}
  />
</DropDownMenu>
`;
